src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -956,7 +954,7 @@ private void CheckMaxReorgRuleViolated(ChainedHeader chainedHeader)
                 {
                     int reorgLength = consensusTip.Height - fork.Height;
 
-                    if (reorgLength > maxReorgLength)
+                    if (fork.Height <= this.finalizedBlockHeight.GetFinalizedBlockHeight())
                     {
                         this.logger.LogTrace("Reorganization of length {0} prevented, maximal reorganization length is {1}, consensus tip is '{2}'.", reorgLength, maxReorgLength, consensusTip);
```

Comment:
It might be nice to have the actual finalise block height in the log as well for comparison.

Also the consensus state is no longer relevant and should not be in the log.

Reply:
Fixed